adminguide.rst

```diff
@@ -74,6 +77,92 @@ with operation differentiated only by configuration.
    Fox prevents Frog from submitting jobs on a cluster with Flux
    as the system resource manager.
 
+Software Components
+===================
+
+Flux was conceived as a resource manager toolkit rather than a monolithic
+project, with the idea to make components like the scheduler replaceable.
+In addition, several parts of flux can be extended with plugins.  At this
+time the primary component types are
+
+broker modules
+  Each broker module runs in its own thread as part of the broker executable,
+  communicating with other components using messages.  Broker modules are
+  dynamically loadable with the :core:man1:`flux-module` command.  Core
+  services like the KVS, job manager, and scheduler are implemented using
+  broker modules.
+
+jobtap plugins
+  The job manager orchestrates a job's life cycle.  Jobtap plugins extend the
+  job manager, arranging for callbacks at different points in the job life
+  cycle, similar to Slurm's SPANK plugins.  Jobtap plugins may be dynamically
```

Comment:
I would remove the comparison to SPANK (since SPANK plugins are more like shell plugins, not jobtap plugins, which would be plugins within the slurmctld)

Reply:
Oops sorry for that misreference.  I'll remove.